aea/components/base.py

```diff
@@ -130,9 +134,184 @@ def load_aea_package(configuration: ComponentConfiguration) -> None:
     author = configuration.author
     package_type_plural = configuration.component_type.to_plural()
     package_name = configuration.name
+    _CheckUsedDependencies(configuration).run_check()
     perform_load_aea_package(dir_, author, package_type_plural, package_name)
 
 
+class _CheckUsedDependencies:
+    """Auxiliary class to keep track of used packages in import statements of package modules."""
+
+    package_type_plural_regex = (
+        rf"({PackageType.AGENT.to_plural()}|"
+        rf"{PackageType.PROTOCOL.to_plural()}|"
+        rf"{PackageType.SKILL.to_plural()}|"
+        rf"{PackageType.CONNECTION.to_plural()}|"
+        rf"{PackageType.CONTRACT.to_plural()}|"
+        rf"{PackageType.SERVICE.to_plural()})"
+    )
+
+    def __init__(self, configuration: ComponentConfiguration) -> None:
+        """Initialize the instance."""
+        self.configuration = configuration
+        enforce(
+            configuration.directory is not None,
+            "input configuration must be associated to a directory",
+            exception_class=ValueError,
+        )
+        self.directory = configuration.directory
+
+    @classmethod
+    def _extract_imported_packages_as_ids(
+        cls, module_content: str
+    ) -> Set[PackageIdPrefix]:
+        """
+        Given a Python module file in form of string, extract all packages being imported.
+
+        E.g. consider an AEA package Python module:
+
+            import packages.open_aea.protocols.signing
+            from packages.fetchai.connections.ledger.connection import LedgerConnection
```

Comment:
What about vendor packages? Don't we need to check for vendor packages as well

Reply:
I think this import pattern works both for non-vendor and vendor packages